src/storage/compaction_checker.cc

```diff
@@ -104,16 +106,23 @@ void CompactionChecker::PickCompactionFiles(const std::string &cf_name) {
     }
 
     if (start_key.empty() || stop_key.empty()) continue;
-    // pick the file which was created more than 2 days
-    if (file_creation_time < static_cast<uint64_t>(now - forceCompactSeconds)) {
-      LOG(INFO) << "[compaction checker] Going to compact the key in file(created more than 2 days): " << iter.first;
-      s = storage_->Compact(&start_key, &stop_key);
-      LOG(INFO) << "[compaction checker] Compact the key in file(created more than 2 days): " << iter.first
+    double delete_ratio = static_cast<double>(deleted_keys) / static_cast<double>(total_keys);
+
+    // pick the file according to force compact policy
+    if (file_creation_time < static_cast<uint64_t>(now - force_compact_file_age) &&
+        delete_ratio >= force_compact_min_ratio) {
+      LOG(INFO) << "[compaction checker] Going to compact the key in file (force compact policy): " << iter.first;
+      auto s = storage_->Compact(&start_key, &stop_key);
+      LOG(INFO) << "[compaction checker] Compact the key in file (force compact policy): " << iter.first
                 << " finished, result: " << s.ToString();
       maxFilesToCompact--;
+      continue;
```

Comment:
@git-hulk @PragmaTwice @xiaobiaozhao I kindly ask you to pay attention to this new `continue` sentence because this will actually change how the `compaction_checker` works.

I add this because if a SST file has been compacted due to the "force compact policy", it should not be considered as a possible candidate with the `best_delete_ratio`. If it is not so skipped, the file may be compacted twice in a single call of `PickCompactionFiles`, from my point of view.

If any of you think this change is not proper please let me know.

Reply:
@ellutionist Thanks for your point out. You're right that is unnecessary to check again.